zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -314,7 +320,44 @@ where
 
             let miner_fee = miner_fee(&mempool_txs);
 
+            // Calling state with `ChainInfo` request for relevant chain data
+            let request = ReadRequest::ChainInfo;
+            let response = state
+                .ready()
+                .and_then(|service| service.call(request))
+                .await
+                .map_err(|error| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                                data: None,
+                            })?;
+
+            let chain_info = match response {
+                ReadResponse::ChainInfo(Some(chain_info)) => chain_info,
+                _ => unreachable!("lets hope for always some until later"),
+            };
+
+            // Get the tip data from the state call
+            let tip_height = chain_info.tip.0;
+            let tip_hash = chain_info.tip.1;
+
             let block_height = (tip_height + 1).expect("tip is far below Height::MAX");
+            let min_time = chain_info.median_time_past.checked_add_signed(Duration::seconds(1))
```

Comment:
Let's write down the consensus rules here, so we are sure we have implemented them correctly.

```suggestion
            // > For each block other than the genesis block , nTime MUST be strictly greater than
            // > the median-time-past of that block.
            // https://zips.z.cash/protocol/protocol.pdf#blockheader
            let min_time = chain_info.median_time_past.checked_add_signed(Duration::seconds(1))
```